src/force_types/force_snap_neigh_impl.h

```diff
@@ -172,15 +172,15 @@ void ForceSNAP<NeighborClass>::compute(System* system, Binning* binning, Neighbo
     const int num_neighs = neighs_i.get_num_neighs();
     if(max_neighs<num_neighs) max_neighs = num_neighs;
   }*/
-  Kokkos::parallel_reduce("ForceSNAP::find_max_neighs",nlocal, FindMaxNumNeighs<t_neigh_list>(neigh_list), Kokkos::Experimental::Max<int>(max_neighs));
+  Kokkos::parallel_reduce("ForceSNAP::find_max_neighs",nlocal, FindMaxNumNeighs<t_neigh_list>(neigh_list), Kokkos::Max<int>(max_neighs));
 
   sna.nmax = max_neighs;
 
   T_INT team_scratch_size = sna.size_team_scratch_arrays();
   T_INT thread_scratch_size = sna.size_thread_scratch_arrays();
 
   //printf("Sizes: %i %i\n",team_scratch_size/1024,thread_scratch_size/1024);
-  int team_size_max = Kokkos::TeamPolicy<>::team_size_max(*this);
+  int team_size_max = Kokkos::TeamPolicy<>(1,1).team_size_max(*this,Kokkos::ParallelForTag());
```

Comment:
This seems really ad-hoc. Can this be made more precise?

Reply:
Agreed. I had mentioned this in that specific commit. Will make this more precise.

Reply:
OK thanks @alanphumphrey. The other changes look good.

Reply:
@stanmoore1 - For clarity, as of Kokkos v3.0, the team_size_max is now a member function of the team policy. The previous variant didn't take all necessary information into account and could result in invalid answers.

In obtaining the team_size_max value, a Kokkos::TeamPolicy<> instance is now created in your code (force_types/force_snap_neigh_impl.h:183), using the default execution space and launching a league with "league_size" teams with "team_size" threads each.

As the team_size_max call is specifically to determine that value, we use Kokkos::AUTO to pick the "team_size" arg. As long as the loop is simpler with only few variables, Kokkos::AUTO should work well. We can then use nlocal for the "league_size" arg (as you've done in force_types/force_snap_neigh_impl.h:192).

Although these changes are more precise and should work well, let me know if this still seems ad-hoc. You may consider slightly reorganizing void ForceSNAP<NeighborClass>::compute based on this input (without the anonymous policy instance). I'm happy to help.

For example, with Nvidia GPUs you might want to consider trying the number SMs for league_size. For say, Kokkos::OpenMP, a reasonable starting point might be a league size of <#_of_cores> and a team size between <1_to_#_of_threads_per_core>. Our group (Uintah AMT runtime system) has experimented with this quite a bit.

This optimal performance balancing act is mostly just about having enough threads to fill to either be compute, memory, or latency bounded.